library/Icingadb/ProvidedHook/X509/Sni.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Icinga\Module\Icingadb\ProvidedHook\X509;
+
+use Icinga\Data\Filter\Filter;
+use Icinga\Module\Icingadb\Common\Auth;
+use Icinga\Module\Icingadb\Common\Database;
+use Icinga\Module\Icingadb\Model\Host;
+use Icinga\Module\X509\Hook\SniHook;
+use ipl\Orm\Compat\FilterProcessor;
+use ipl\Web\Filter\QueryString;
+
+class Sni extends SniHook
+{
+    use Auth;
+    use Database;
+
+    /**
+     * @inheritDoc
+     */
+    public function getHosts(Filter $filter = null)
+    {
+        $queryHost = Host::on($this->getDb());
+
+        $queryHost->getSelectBase();
+
+        $hostStatusCols = [
+            'host_name'      => 'name',
+            'host_address'   => 'address',
+            'host_address6'  => 'address6'
+        ];
+
+        $queryHost = $queryHost->columns($hostStatusCols);
+
+        $this->applyRestrictions($queryHost);
+
+        $filter = Filter::where('name', 'dummy-1');
+        if ($filter !== null) {
+            $queryString = $filter->toQueryString();
+            $filterCondition = QueryString::parse($queryString);
+            FilterProcessor::apply($filterCondition, $queryHost);
```

Comment:
The `FilterProcessor` class is now not necessary anymore. Use this instead:

```suggestion
            $queryHost->filter($filterCondition);
```

Reply:
It makes sense. I will make the changes.